ExampleMod/Content/Tiles/ExampleCrystal.cs

```diff
@@ -0,0 +1,116 @@
+﻿using ExampleMod.Content.Dusts;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.ObjectData;
+
+namespace ExampleMod.Content.Tiles
+{
+    internal class ExampleCrystal : ModTile
+    {
+        public override void SetStaticDefaults()
+        {
+            TileObjectData.newTile.Width = 1;
+            TileObjectData.newTile.Height = 1;
+            TileObjectData.newTile.CoordinateWidth = 16; // Width of tile on spritesheet
+            TileObjectData.newTile.CoordinateHeights = new int[] { 16 }; // Sets an offset
+            TileObjectData.newTile.CoordinatePadding = 2; // Sets an "empty space" that should be between each 16x16 sprite in spritesheet
+            TileObjectData.newTile.StyleHorizontal = true;
+            TileObjectData.addTile(Type);
+            Main.tileFrameImportant[Type] = true;
+            Main.tileObsidianKill[Type] = true;
+            Main.tileShine2[Type] = true;
+            Main.tileShine[Type] = 500;
+            Main.tileSpelunker[Type] = true;
+            ItemType = ModContent.ItemType<ExampleCrystalItem>();
+            DustType = ModContent.DustType<Sparkle>();
+        }
+
+        public override ushort GetMapOption(int i, int j) 
+			=> (ushort)(Main.tile[i, j].TileFrameX / 18); // this override is redundant, but in case if you want more map names option ig, then you can leave it alone?
+
+        public override bool TileFrame(int i, int j, ref bool resetFrame, ref bool noBreak)
+        {
+            Main.tile[i, j].TileType = TileID.Crystals;
+            WorldGen.TileFrame(i, j, resetFrame, noBreak); // applies vanilla Crystal Shards behavior
+            Main.tile[i, j].TileType = Type;
+            return false;
+        }
+    }
+
+    // We need a GlobalTile class to make sure our crystal can randomly spawn in the world
+	// The logic is contained in the RandomUpdate function
+    internal class ExampleCrystalGrow : GlobalTile
+    {
+        public override void RandomUpdate(int i, int j, int type)
+        {
+			// The crystal will grow only if: the world is in Hardmode, the tile is Stone, it's in the rock layer
+			// and with same chance as the vanilla Crystal Shards have of spawning.
+            if (Main.hardMode && type == TileID.Stone && j > Main.rockLayer && WorldGen.genRand.NextBool(110))       
+            {
+				// Randomly generate the direction the crystal spawns in
+                int coordinateChance = WorldGen.genRand.Next(4);
+                int additionalX = 0;
+                int additionalY = 0;
+				
+                if (coordinateChance == 0)
+                {
+                    additionalX = -1;
+                    additionalY = -1;
+                }
+                else if (coordinateChance == 1)
+                {
+                    additionalX = 1;
+                    additionalY = 1;
+                }
```

Comment:
I've added some better comments and made the variables clearer in the tile placement section.
However, I am still bothered by this section. We're generating a random choice out of 4 options, but we only use 2 of them and in a weird way. It makes more sense to me if there are 4 options of using the coordinateChance, because the crystal could grown on any of the 4 sides a tile has (top, right, bottom, left). So the appropriate thing to do here would be do end up with 4 combinations:
```
x = 1, y = 0 (right)
x = -1, y = 0 (left)
x = 0, y = 1 (bottom)
x = 0, y = -1 (top)
```
I will leave this update to you as a tiny challenge :)

Reply:
i took spreading thing from vanilla, so idk why its like that 🤷